lib/components/Breadcrumbs/base/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { BreadcrumbProps, LinkProps } from "../types";
+import { Fragment, useEffect, useState } from "react";
+import { maxLinks } from "./constants";
+import "./breadcrumbs.scss";
+import useBreakpoints from "@hooks/useBreakpoints";
+import clsx from "clsx";
+import Link from "@components/Link";
+
+export function Base({ size = "sm", links = [], className }: BreadcrumbProps) {
+    const [renderLinks, setRenderLinks] = useState<LinkProps[]>([]);
+    const [dropdownLinks, setDropdownLinks] = useState<LinkProps[]>([]);
+
+    const isLastItem = (key: number) => renderLinks.length - 1 > key;
+    const linksLen = links.length;
+    const { isMobile } = useBreakpoints();
+    const hasExtra = isMobile && maxLinks < linksLen;
+
+    useEffect(() => {
+        setRenderLinks(links);
+
+        if (hasExtra) {
+            const limitedLinks = [
+                links[0],
+                { content: "...", href: "" },
+                links[linksLen - 2],
+                links[linksLen - 1],
+            ];
+
+            setRenderLinks(limitedLinks);
+
+            const remainingLinks: LinkProps[] = links.filter(
+                (item) => !new Set(limitedLinks).has(item),
+            );
+
+            setDropdownLinks(remainingLinks);
+        }
+    }, [links]);
+
+    return (
+        <div className={clsx("breadcrumb--container", className)}>
```

Comment:
Can we make sure all components container has a classname starts with `quill-` so we can handle them accordingly in the future.